recipes/qpdf/all/conanfile.py

```diff
@@ -0,0 +1,164 @@
+        return {
+            "gcc": "5",
+            "clang": "3.4",
+            "apple-clang": "10",
+        }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        # https://qpdf.readthedocs.io/en/stable/installation.html#basic-dependencies
+        self.requires("zlib/1.2.12")
+        if self.options.with_crypto == "openssl":
+            self.requires("openssl/1.1.1q")
+        elif self.options.with_crypto == "gnutls":
+            raise ConanInvalidConfiguration("GnuTLS is not available in Conan Center yet.")
+        if self.options.with_jpeg == "libjpeg":
+            self.requires("libjpeg/9e")
+        elif self.options.with_jpeg == "libjpeg-turbo":
+            self.requires("libjpeg-turbo/2.1.4")
+        elif self.options.with_jpeg == "mozjpeg":
+            self.requires("mozjpeg/4.0.0")
+
+
+    def validate(self):
+        if self.info.settings.compiler.cppstd:
+            check_min_cppstd(self, self._minimum_cpp_standard)
+        if is_msvc(self):
+            check_min_vs(self, "150")
+        else:
+            minimum_version = self._compilers_minimum_version.get(str(self.info.settings.compiler), False)
+            if minimum_version and Version(self.info.settings.compiler.version) < minimum_version:
+                raise ConanInvalidConfiguration(f"{self.ref} requires C++{self._minimum_cpp_standard}, which your compiler does not support.")
+
+    def build_requirements(self):
+        self.tool_requires("cmake/3.24.1")
+        self.tool_requires("pkgconf/1.9.3")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version],
+                  destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["BUILD_STATIC_LIBS"] = not self.options.shared
+        # https://qpdf.readthedocs.io/en/latest/installation.html#build-time-crypto-selection
+        tc.variables["USE_IMPLICIT_CRYPTO"] = False
+        tc.cache_variables["CMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP"] = True
+        if self.options.with_crypto == "native":
+            tc.variables["REQUIRE_CRYPTO_NATIVE"] = True
+            tc.variables["REQUIRE_CRYPTO_GNUTLS"] = False
+            tc.variables["REQUIRE_CRYPTO_OPENSSL"] = False
+        elif self.options.with_crypto == "openssl":
+            tc.variables["REQUIRE_CRYPTO_NATIVE"] = False
+            tc.variables["REQUIRE_CRYPTO_GNUTLS"] = False
+            tc.variables["REQUIRE_CRYPTO_OPENSSL"] = True
+        tc.generate()
+        tc = PkgConfigDeps(self)
+        tc.generate()
```

Comment:
Good question. I think we would only loose the print-out during detection. But those detected are not anymore used after my cmake patches. I don't know if it is better to have the found-printouts, that later-on the missing dependency prints are not confusing for other looking in the build-log. What would you prefer?

Reply:
I think the missing print-out will be easier to track. Someone looking at the errors is likely to think the dependencies are coming from `pkg-config` instead of CMake modules, which could lead to a lot of wasted time when debugging. Using `find_package` in your patches should provide at least some output to the console, too.

Reply:
Ok i see your point and i go with it.

I can confirm on local builds, that it is not necessary anymore. So i will remove it. Nevertheless, i try to add an explanatory comment on why we don't go for pkgconfig, even when the qpdf uses that by default